### PR TITLE
PLAT-372 convert mds-api-* to jest

### PR DESCRIPTION
**Pull request by nessur**
## 📚 Purpose
*[convert all tests in mds-api-* to jest from mocha/sinon/unitjs ]*

## 👌 Resolves:
- [x] ✨ PLAT-372 

## 📦 Impacts:
*[mds-api-helpers]*
*[mds-api-authorizer]*
*[mds-api-server]*

## ✅ PR Checklist
- [x] mds-api-authorizer
- [x] mds-api-helpers
- [x] mds-api-server

### CODE REVIEW

File: packages/mds-api-helpers/package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
   ],
   "scripts": {
     "build": "tsc --build tsconfig.build.json",
-    "test": "DOTENV_CONFIG_PATH=../../.env nyc --use-spawn-wrap --lines 40  ts-mocha --project ../../tsconfig.json"
+    "test": "DOTENV_CONFIG_PATH=../../.env jest --runInBand --forceExit "
```
Comment (avatarneil) [inline packages/mds-api-helpers/package.json:17]: Does this need `--forceExit`? Looks like the mocha tests didn't. I think we should try to isolate usage of this only to cases where it's super necessary, as it *usually* indicates there's a nasty dangling promise bug.

File: packages/mds-api-helpers/tests/index.spec.ts [hunk 2]
```diff
@@ -14,13 +14,27 @@
...
-import test from 'unit.js'
-import { parseRequest } from '../index'
+import { asJsonApiLinks, parseRequest } from '../index'
 
 const arraysEqual = <T>(array1?: T[], array2?: T[]) => JSON.stringify(array1) === JSON.stringify(array2)
 
 describe('mds-api-helpers Tests', () => {
+  describe('asJsonApiLinks test', () => {
+    it('outputs links', () => {
+      const uuid1 = uuid()
```
Comment (avatarneil) [inline packages/mds-api-helpers/tests/index.spec.ts:27]: Super dumb nit, but putting numbers in variables tingles my spidey sense tingle funny (feels like the first step towards having `const foo1; const foo2; ... const foon`). Could we just call this `provider_id` or something?

File: packages/mds-api-server/package.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
   ],
   "scripts": {
     "build": "tsc --build tsconfig.build.json",
-    "test": "DOTENV_CONFIG_PATH=../../.env nyc --use-spawn-wrap ts-mocha --project ../../tsconfig.json"
+    "test": "DOTENV_CONFIG_PATH=../../.env jest --runInBand --forceExit "
```
Comment (avatarneil) [inline packages/mds-api-server/package.json:16]: Same nit about `forceExit`, would like to only have it if necessary.

File: packages/mds-api-server/tests/index.spec.ts [hunk 4]
```diff
@@ -44,163 +43,133 @@ const request = supertest(api)
...
     try {
-      test.value(server.keepAliveTimeout).is(Number(process.env.HTTP_KEEP_ALIVE_TIMEOUT))
-    } catch (err) {
-      error = err
+      expect(server.keepAliveTimeout).toStrictEqual(Number(process.env.HTTP_KEEP_ALIVE_TIMEOUT))
+    } finally {
+      server.close()
     }
-    server.close()
-    done(error)
```
Comment (avatarneil) [inline packages/mds-api-server/tests/index.spec.ts:136]: I have no idea why this needed to be a try/catch to begin with, or a try/finally now. Could this just not have any `try` wrapping?

**Review by avatarneil — COMMENTED** [review]
Looks really good!!! A couple minor things I'd like to see addressed, though.

**Review by PeterSP — APPROVED** [review]

**Review by avatarneil — APPROVED** [review]
image

**Pull request merged**